riscv/src/register/tests/read_only_csr.rs

```diff
@@ -0,0 +1,127 @@
+use crate::result::{Error, Result};
+
+read_only_csr! {
+    /// test CSR register type
+    Mtest: 0x000,
+    mask: 0b1111_1111_1111,
+}
+
+read_only_csr_field! {
+    Mtest,
+    /// test single-bit field
+    single: 0,
+}
+
+read_only_csr_field! {
+    Mtest,
+    /// multiple single-bit field range
+    multi_range,
+    /// try-getter multiple single-bit field range
+    try_multi_range,
+    range: 1..=3,
+}
+
+read_only_csr_field! {
+    Mtest,
+    /// multi-bit field
+    multi_field: [4:7],
+}
+
+csr_field_enum! {
+    /// field enum type with valid field variants
+    MtestFieldEnum {
+        range: [8:11],
+        default: Field1,
+        Field1 = 1,
+        Field2 = 2,
+        Field3 = 3,
+        Field4 = 15,
+    }
+}
+
+read_only_csr_field! {
+    Mtest,
+    /// multi-bit field
+    field_enum,
+    /// try-getter multi-bit field
+    try_field_enum,
+    MtestFieldEnum,
+    range: [8:11],
+}
+
+// we don't test the `read` function, we are only testing in-memory functions.
+#[allow(unused)]
+pub fn _read_csr() -> Mtest {
+    read()
+}
+
+#[allow(unused)]
+pub fn _try_read_csr() -> Result<Mtest> {
+    try_read()
+}
+
+#[test]
+fn test_mtest_read_only() {
+    let mut mtest = Mtest::from_bits(0);
+
+    assert_eq!(mtest.bitmask(), Mtest::BITMASK);
+    assert_eq!(mtest.bits(), 0);
+
+    // check that single bit field getter/setters work.
+    assert_eq!(mtest.single(), false);
```

Comment:
Clippy complains at my machine about these kinds of assertions. Should look like:

```suggestion
    assert!(!mtest.single());
```